samples/magic/az_ml_magic.py

```diff
@@ -0,0 +1,300 @@
+from __future__ import print_function
+from IPython.core.magic import (Magics, magics_class, line_magic, cell_magic)
+from IPython.utils.ipstruct import Struct
+import os
+import argparse
+import shlex
+import re
+
+from cStringIO import StringIO
+import sys
+
+
+class Capturing(list):
+    def __enter__(self):
+        self._stdout = sys.stdout
+        sys.stdout = self._stringio = StringIO()
+        return self
+
+    def __exit__(self, *args):
+        self.extend(self._stringio.getvalue().splitlines())
+        del self._stringio  # free up some memory
+        sys.stdout = self._stdout
+
+
+@magics_class
+class AMLHelpers(Magics):
+    @staticmethod
+    def print_and_update_env(k, v):
+        os.environ[k] = v
+        print(' os.environ["{}"]="{}"'.format(k, v))
+        return 'export {}={}'.format(k, v)
+
+    @cell_magic
+    def save_file(self, parameter_s='', cell=None):
+        opts, arg_str = self.parse_options(parameter_s, 'f:', list_all=True, posix=False)
+        if cell is not None:
+            arg_str += '\n' + cell
+
+        return self._save_file(arg_str, opts, self.shell.user_ns)
+
+    @line_magic
+    def list_subs(self, line):
+        from azure.cli.core._profile import Profile
+        try:
+            from azure.cli.core.util import CLIError
+        except ImportError:
+            from azure.cli.core._util import CLIError
+        self._redirect_logging('az.azure.cli.core._profile')
+        profile = Profile()
```

Comment:
Is this what prompts for login? Can you add a comment describing what Profile() does?